relay-general/src/pii/config.rs

```diff
@@ -8,10 +8,33 @@ use serde::{de::Error, Deserialize, Deserializer, Serialize, Serializer};
 
 use crate::pii::{CompiledPiiConfig, Redaction};
 use crate::processor::SelectorSpec;
+use failure::Fail;
+
+#[derive(Clone, Debug, Fail)]
+pub enum PiiConfigError {
+    #[fail(display = "could not parse regex")]
+    RegexError(#[cause] regex::Error),
+}
 
 /// A regex pattern for text replacement.
 #[derive(Clone)]
-pub struct Pattern(pub Regex);
+pub struct Pattern(Regex);
+
+impl Pattern {
+    pub fn new(s: &str, case_insensitive: bool) -> Result<Self, PiiConfigError> {
+        let regex = RegexBuilder::new(s)
+            .size_limit(262_144)
```

Comment:
Apply the same size limit to every `Pattern` that we create.

Reply:
Why that size limit specifically?

Reply:
Also: should we make the number a constant?